### PR TITLE
support modules with inner paths

### PR DESCRIPTION
**Pull request by carmelc**
* Support `exports` field
* Add in addition to `unpkg` support `jsdeliver`
* Support nested modules in one package `package/module/file.ts`
* Support `types` `typings` `import` `require`

### CODE REVIEW

File: __tests__/e2e/common-libs-e2e.fixed-versions.spec.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import tempy from 'tempy';
+import { bundle } from '../../src';
+import fs from 'fs';
+
+describe('Common Libs - Fixed versions', () => {
```
Comment (yurynix) [inline __tests__/e2e/common-libs-e2e.fixed-versions.spec.ts:5]: What `Fixed` means? 🤔 I would clarify what it means or remove it if that makes sense.

File: __tests__/e2e/site-sdk-e2e.fixed-versions.spec.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      
+      queryParams.add({key: 'value'}).then(console.log);
+      currentMember.getMember().then(console.log);
+      siteBookings.getServiceAvailability('id').then(console.log);
+    `;
+
+    expect(validateTypescript(fileContent, tempDir).isValid).toBe(true);
+  });
+
+  it('should create typescript error when the code is not valid', async () => {
```
Comment (yurynix) [inline __tests__/e2e/site-sdk-e2e.fixed-versions.spec.ts:60]: love that

File: __tests__/e2e/utils.ts [hunk 3]
```diff
@@ -62,10 +69,11 @@ export interface MockedFetch extends Fetch {
 }
 
 jest.mock('node-fetch', () => {
+  const { JSDELIVR_BASE, UNPKG_BASE, JSDELIVR_API_BASE } = require('../../src/consts');
```
Comment (yurynix) [inline __tests__/e2e/utils.ts]: maybe the mock is better in a separate file instead of in `utils`? dunno 🤷‍♂️

File: src/create-entry-file.ts [hunk 4]
```diff
@@ -90,26 +104,33 @@ const getTypesImportForModule = (packageName: string, pkgJsonData: PackageJson,
   return result;
 };
 
-export const createPackageTypes = async (packageName: string, packageVersion: string, fileFetcher: (filePath: string, content?: string) => Promise<string>) => {
-  const url = `https://unpkg.com/${packageName}@${packageVersion}/?meta`;
+export const createPackageTypes = async (packageName: string, packageVersion: string, fileFetcher: (filePath: string, content?: string) => Promise<string>, cdn: CDN) => {
+  const url = cdn === CDN.UNPKG ?
```
Comment (yurynix) [inline src/create-entry-file.ts]: I would do something like that:
```es6
const CDNs = {
   UNPKG: {
     baseUrl: '',
     packageFileListURL = (name, version) => `${UNPKG_BASE}${packageName}@${packageVersion}/?meta`,
     collectPackageJsonUrls: (node: FlatFiles) => ...,
   },
   JSDELIVER: {
   ...
   }
}
```

and then do:
```es6
const cdn = CDNs[cdnName];
const url = cdn.packageFileListURL('lodash', '1.2.3');
....
const pathsToRead = cdn.collectPackageJsonUrls(packageMeta);
```
etc. etc.

**Review by yurynix — APPROVED** [review]
Love that alot, thanks for working on that 🙏

File: __tests__/e2e/common-libs-e2e.fixed-versions.spec.ts (hunk 1 above)
Comment (carmelc) [reply __tests__/e2e/common-libs-e2e.fixed-versions.spec.ts:5]: fixed as in static, hard coded in appose to latest,
when you use "1.2.3" in package json it is a fixed version.

File: __tests__/e2e/utils.ts (hunk 3 above)
Comment (carmelc) [reply __tests__/e2e/utils.ts]: Did better, there is a test CDN implementation which is removing the need to do any fetch mock

**Pull request merged**